### PR TITLE
fix #151: use serialize instead of json_encode

### LINKED ISSUE #151: Cannot send email
Hope that I can get answer here
I have already install a new cakephp dan cakephp-queue on my machine. But it didn't work when queueworker send the email.

I always get this error :

2017-10-16 07:08:44 Warning: Warning (2): call_user_func_array() expects parameter 1 to be a valid callback, class 'Cake\Mailer\Email' does not have a method 'viewConfig' in [D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE\src\Error\BaseErrorHandler.php, line 153
Queue\Shell\Task\QueueEmailTask::run() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110
Queue\Shell\QueueShell::runworker() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php, line 169
Cake\Console\Shell::runCommand() - CORE\src\Console\Shell.php, line 494
Cake\Console\CommandRunner::run() - CORE\src\Console\CommandRunner.php, line 141
[main] - ROOT\bin\cake.php, line 12

2017-10-16 07:08:44 Warning: Warning (2): call_user_func_array() expects parameter 1 to be a valid callback, class 'Cake\Mailer\Email' does not have a method '_to' in [D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE\src\Error\BaseErrorHandler.php, line 153
Queue\Shell\Task\QueueEmailTask::run() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110
Queue\Shell\QueueShell::runworker() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php, line 169
Cake\Console\Shell::runCommand() - CORE\src\Console\Shell.php, line 494
Cake\Console\CommandRunner::run() - CORE\src\Console\CommandRunner.php, line 141
[main] - ROOT\bin\cake.php, line 12

2017-10-16 07:08:44 Warning: Warning (2): call_user_func_array() expects parameter 1 to be a valid callback, class 'Cake\Mailer\Email' does not have a method '_from' in [D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE\src\Error\BaseErrorHandler.php, line 153
Queue\Shell\Task\QueueEmailTask::run() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110
Queue\Shell\QueueShell::runworker() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php, line 169
Cake\Console\Shell::runCommand() - CORE\src\Console\Shell.php, line 494
Cake\Console\CommandRunner::run() - CORE\src\Console\CommandRunner.php, line 141
[main] - ROOT\bin\cake.php, line 12

2017-10-16 07:08:44 Warning: Warning (2): call_user_func_array() expects parameter 1 to be a valid callback, class 'Cake\Mailer\Email' does not have a method '_subject' in [D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE\src\Error\BaseErrorHandler.php, line 153
Queue\Shell\Task\QueueEmailTask::run() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110
Queue\Shell\QueueShell::runworker() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php, line 169
Cake\Console\Shell::runCommand() - CORE\src\Console\Shell.php, line 494
Cake\Console\CommandRunner::run() - CORE\src\Console\CommandRunner.php, line 141
[main] - ROOT\bin\cake.php, line 12

2017-10-16 07:08:44 Warning: Warning (2): call_user_func_array() expects parameter 1 to be a valid callback, class 'Cake\Mailer\Email' does not have a method '_emailFormat' in [D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE\src\Error\BaseErrorHandler.php, line 153
Queue\Shell\Task\QueueEmailTask::run() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110
Queue\Shell\QueueShell::runworker() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php, line 169
Cake\Console\Shell::runCommand() - CORE\src\Console\Shell.php, line 494
Cake\Console\CommandRunner::run() - CORE\src\Console\CommandRunner.php, line 141
[main] - ROOT\bin\cake.php, line 12

2017-10-16 07:08:44 Warning: Warning (2): call_user_func_array() expects parameter 1 to be a valid callback, class 'Cake\Mailer\Email' does not have a method '_emailPattern' in [D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE\src\Error\BaseErrorHandler.php, line 153
Queue\Shell\Task\QueueEmailTask::run() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110
Queue\Shell\QueueShell::runworker() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php, line 169
Cake\Console\Shell::runCommand() - CORE\src\Console\Shell.php, line 494
Cake\Console\CommandRunner::run() - CORE\src\Console\CommandRunner.php, line 141
[main] - ROOT\bin\cake.php, line 12

2017-10-16 07:08:44 Warning: Warning (2): call_user_func_array() expects parameter 1 to be a valid callback, class 'Cake\Mailer\Email' does not have a method '_domain' in [D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE\src\Error\BaseErrorHandler.php, line 153
Queue\Shell\Task\QueueEmailTask::run() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110
Queue\Shell\QueueShell::runworker() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php, line 169
Cake\Console\Shell::runCommand() - CORE\src\Console\Shell.php, line 494
Cake\Console\CommandRunner::run() - CORE\src\Console\CommandRunner.php, line 141
[main] - ROOT\bin\cake.php, line 12

2017-10-16 07:08:44 Warning: Warning (2): call_user_func_array() expects parameter 1 to be a valid callback, class 'Cake\Mailer\Email' does not have a method '_messageId' in [D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE\src\Error\BaseErrorHandler.php, line 153
Queue\Shell\Task\QueueEmailTask::run() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110
Queue\Shell\QueueShell::runworker() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php, line 169
Cake\Console\Shell::runCommand() - CORE\src\Console\Shell.php, line 494
Cake\Console\CommandRunner::run() - CORE\src\Console\CommandRunner.php, line 141
[main] - ROOT\bin\cake.php, line 12

2017-10-16 07:08:44 Warning: Warning (2): call_user_func_array() expects parameter 1 to be a valid callback, class 'Cake\Mailer\Email' does not have a method '_appCharset' in [D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE\src\Error\BaseErrorHandler.php, line 153
Queue\Shell\Task\QueueEmailTask::run() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php, line 110
Queue\Shell\QueueShell::runworker() - ROOT\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php, line 169
Cake\Console\Shell::runCommand() - CORE\src\Console\Shell.php, line 494
Cake\Console\CommandRunner::run() - CORE\src\Console\CommandRunner.php, line 141
[main] - ROOT\bin\cake.php, line 12

2017-10-16 07:08:44 Error: QueueEmail
BadMethodCallException: You need specify one destination on to, cc or bcc.
#0 D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\Task\QueueEmailTask.php(126): Cake\Mailer\Email->send(NULL)
#1 D:\xampp\htdocs\e-commerce\vendor\dereuromark\cakephp-queue\src\Shell\QueueShell.php(169): Queue\Shell\Task\QueueEmailTask->run(Array, 8)
#2 D:\xampp\htdocs\e-commerce\vendor\cakephp\cakephp\src\Console\Shell.php(494): Queue\Shell\QueueShell->runworker()
#3 D:\xampp\htdocs\e-commerce\vendor\cakephp\cakephp\src\Console\CommandRunner.php(141): Cake\Console\Shell->runCommand(Array, true)
#4 D:\xampp\htdocs\e-commerce\bin\cake.php(12): Cake\Console\CommandRunner->run(Array)
#5 {main}

### PR DESCRIPTION
**Pull request by stmeyer**
Closes https://github.com/dereuromark/cakephp-queue/issues/151

### CODE REVIEW

**Comment by dereuromark** [comment]
The tests seem to need adjusting, instead of null it now is false returned for unserialize.

**Comment by stmeyer** [comment]
Yes, it works now. In the first change https://github.com/dereuromark/cakephp-queue/commit/7ab4d157afb8178189116f4adb0c631bff19a99b#diff-4fc1d72592309bfca7e3222134880c0cL327 is was asserted NULL.

File: src/Shell/QueueShell.php [hunk 1]
```diff
@@ -163,7 +163,7 @@ public function runworker() {
 				$taskname = 'Queue' . $queuedTask['job_type'];
 
 				try {
-					$data = json_decode($queuedTask['data'], true);
+					$data = unserialize($queuedTask['data']);
```
Comment (dereuromark) [inline src/Shell/QueueShell.php]: This broke master, the change here was unrelated and invalid :/

File: src/Model/Table/QueuedJobsTable.php [hunk 2]
```diff
@@ -102,7 +102,7 @@ public function initConfig() {
 	public function createJob($jobName, array $data = null, array $config = []) {
 		$queuedJob = [
 			'job_type' => $jobName,
-			'data' => is_array($data) ? json_encode($data) : null,
+			'data' => is_array($data) ? serialize($data) : null,
```
Comment (dereuromark) [inline src/Model/Table/QueuedJobsTable.php]: same here

File: src/Shell/QueueShell.php (hunk 1 above)
Comment (dereuromark) [reply src/Shell/QueueShell.php]: I was kind of under the impression we only changed the email task here for it, not changing all of them.. not sure if thats good or not..

**Pull request merged**